Modules/Sources/Yosemite/Model/Storage/PersistedProduct+Conversions.swift

```diff
@@ -0,0 +1,97 @@
+import Foundation
+import Storage
+
+// MARK: - PersistedProduct Conversions
+extension PersistedProduct {
+    public init(from posProduct: POSProduct) {
+        self.init(
+            id: posProduct.productID,
+            siteID: posProduct.siteID,
+            name: posProduct.name,
+            productTypeKey: posProduct.productTypeKey,
+            fullDescription: posProduct.fullDescription,
+            shortDescription: posProduct.shortDescription,
+            sku: posProduct.sku,
+            globalUniqueID: posProduct.globalUniqueID,
+            price: posProduct.price,
+            downloadable: posProduct.downloadable,
+            parentID: posProduct.parentID,
+            manageStock: posProduct.manageStock,
+            stockQuantity: posProduct.stockQuantity,
+            stockStatusKey: posProduct.stockStatusKey
+        )
+    }
+
+    public func toPOSProduct(images: [ProductImage] = [], attributes: [ProductAttribute] = []) -> POSProduct {
+        return POSProduct(
+            siteID: siteID,
+            productID: id,
+            name: name,
+            productTypeKey: productTypeKey,
+            fullDescription: fullDescription,
+            shortDescription: shortDescription,
+            sku: sku,
+            globalUniqueID: globalUniqueID,
+            price: price,
+            downloadable: downloadable,
+            parentID: parentID,
+            images: images,
+            attributes: attributes,
+            manageStock: manageStock,
+            stockQuantity: stockQuantity,
+            stockStatusKey: stockStatusKey
+        )
+    }
+}
+
+// MARK: - PersistedProductAttribute Conversions
+extension PersistedProductAttribute {
+    public init(from productAttribute: ProductAttribute, productID: Int64) {
+        self.init(
+            productID: productID,
+            name: productAttribute.name,
+            position: Int64(productAttribute.position),
+            visible: productAttribute.visible,
+            variation: productAttribute.variation,
+            options: productAttribute.options
+        )
+    }
+
+    public func toProductAttribute(siteID: Int64) -> ProductAttribute {
+        return ProductAttribute(
+            siteID: siteID,
+            attributeID: 0,
```

Comment:
if we decide to set the remote ID in a separate column (https://github.com/woocommerce/woocommerce-ios/pull/16047#issuecomment-3240626369), this can be set. For now, do we want to set this with `id ?? 0` like in the variation attribute?

Reply:
Done in f822a09

The risk (which was already worse than it is now) is that if we call `toProductAttribute` on a global attribute which hasn't been saved, we'll get 0 instead of nil, which suggests a local attribute. But then only local attributes are properly handled so far anyway.